src/main/java/io/strimzi/kafka/metrics/PrometheusCollector.java

```diff
@@ -30,43 +33,133 @@
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
- * Prometheus Collector to store and export metrics retrieved by {@link YammerPrometheusMetricsReporter}.
+ * Prometheus Collector to store and export metrics retrieved by {@link KafkaPrometheusMetricsReporter}
+ * and {@link YammerPrometheusMetricsReporter}.
  */
 @SuppressWarnings("ClassFanOutComplexity")
-public class YammerMetricsCollector implements MultiCollector {
+public class PrometheusCollector implements MultiCollector {
 
-    private static final Logger LOG = LoggerFactory.getLogger(YammerMetricsCollector.class);
+    private static final Logger LOG = LoggerFactory.getLogger(PrometheusCollector.class);
+    private static final AtomicBoolean REGISTERED = new AtomicBoolean(false);
+    private static final PrometheusCollector INSTANCE = new PrometheusCollector();
     private static final List<Double> QUANTILES = Arrays.asList(0.50, 0.75, 0.95, 0.98, 0.99, 0.999);
 
-    private final Map<MetricName, MetricWrapper> metrics;
+    private final Map<MetricName, MetricWrapper> kafkaMetrics = new ConcurrentHashMap<>();
+    private final Map<com.yammer.metrics.core.MetricName, MetricWrapper> yammerMetrics = new ConcurrentHashMap<>();
 
     /**
-     * Constructor
+     * Add a Kafka metric to be collected.
+     *
+     * @param name The name of the Kafka metric to add.
+     * @param metric The Kafka metric to add.
+     */
+    public void addKafkaMetric(MetricName name, MetricWrapper metric) {
+        kafkaMetrics.put(name, metric);
+    }
+
+    /**
+     * Remove a Kafka metric from collection.
+     *
+     * @param name The name of Kafka metric to remove.
+     */
+    public void removeKafkaMetric(MetricName name) {
+        kafkaMetrics.remove(name);
+    }
+
+    /**
+     * Add a Yammer metric to be collected.
+     *
+     * @param name The name of the Yammer metric to add.
+     * @param metric The Yammer metric to add.
+     */
+    public void addYammerMetric(com.yammer.metrics.core.MetricName name, MetricWrapper metric) {
+        yammerMetrics.put(name, metric);
+    }
+
+    /**
+     * Remove a Yammer metric from collection.
+     *
+     * @param name The name of the Yammer metric to remove.
+     */
+    public void removeYammerMetric(com.yammer.metrics.core.MetricName name) {
+        yammerMetrics.remove(name);
+    }
+
+    /**
+     * Register this Collector with the provided Prometheus registry
+     * @param registry The Prometheus registry
+     * @return The PrometheusCollector instance
      */
-    public YammerMetricsCollector() {
-        this.metrics = new ConcurrentHashMap<>();
+    public static PrometheusCollector register(PrometheusRegistry registry) {
```

Comment:
Code smell - single pattern without a private constructor.
I wonder if you shifted the responsibility for the singleton to a separate class (`CollectorHolder`)

Reply:
The only reason I did not add a private constructor is for tests. It's useful to get unique instances instead of the global collector so tests are independent. We can add a package-only constructor.

Reply:
I've added a package only constructor in https://github.com/strimzi/metrics-reporter/pull/46/commits/79dd1950937c5b6fa9d4d9f1cd22031209b22468